src/snyk/common/languageServer/lsExecutable.ts

```diff
@@ -0,0 +1,35 @@
+import path from 'path';
+import { Checksum } from '../../cli/checksum';
+import { Platform } from '../../common/platform';
+import { LsSupportedPlatform } from './supportedPlatforms';
+
+export class LsExecutable {
+  // If values updated, `.vscodeignore` to be changed.
+  public static filenameSuffixes: Record<LsSupportedPlatform, string> = {
+    linux386: 'linux_386',
+    linuxAmd64: 'linux_amd64',
+    linuxArm64: 'linux_arm64',
+    windows386: `windows_386.exe`,
+    windowsAmd64: 'windows_amd64.exe',
+    darwinAmd64: 'darwin_amd64',
+    darwinArm64: 'darwin_arm64',
+  };
+
+  constructor(public readonly version: string, public readonly checksum: Checksum) {}
+
+  static getFilename(platform: LsSupportedPlatform): string {
+    return this.filenameSuffixes[platform];
+  }
+
+  static getPath(extensionDir: string, customPath?: string): string {
+    if (customPath) {
+      return customPath;
+    }
+
+    const platform = Platform.getCurrentWithArch();
+    const fileName = LsExecutable.getFilename(platform);
+    const lsPath = path.join(extensionDir, fileName);
```

Comment:
This is wrong, as the language server will be stored inside of XDG dir to reuse across different IDEs. See 5th scenario in the ticket description.

Reply:
This was to keep it consistent with the CLI implementation, but also, do we want to centralise LS like this? Could having separate IDEs manage the same binary lead to unexpected issues? e.g. what would happen is an IDE is already open and a separate IDE gets opened and updates LS?

Reply:
That was the decision we made when starting with LS integration in Eclipse. You can cross verify by @bastiandoetsch. Your points are totally valid though, maybe it's not that likely that both updates start 100% simultaneously. And update generally should create a lock file, that would prevent other IDEs from starting their own update.

Reply:
Happy to revert it if that's what we want, this change was raised during the last sprint after seeing how it was implemented for the CLI; and all seemed ok with it.